src/components/SimpleLegend/SimpleLegend.tsx

```diff
@@ -2,25 +2,45 @@ import { Text } from '@konturio/ui-kit';
 import { HexIcon } from '~components/SimpleLegend/icons/HexIcon';
 import { CircleIcon } from '~components/SimpleLegend/icons/CircleIcon';
 import { SquareIcon } from '~components/SimpleLegend/icons/SquareIcon';
+import { LetterIcon } from '~components/SimpleLegend/icons/LetterIcon';
 import s from './SimpleLegend.module.css';
 import type { SimpleLegend as SimpleLegendType } from '~core/logical_layers/types/legends';
 import type { CSSProperties } from 'react';
 
 type Flatten<Type> = Type extends Array<infer Item> ? Item : Type;
 
-function icon(
-  type: string,
-  styles: Flatten<SimpleLegendType['steps']>['style'],
-  fill?: string,
-  stroke?: string,
-) {
-  if (!type) return null;
-  if (type === 'hex')
-    return <HexIcon styles={styles} size="small" fill={fill} stroke={stroke} />;
-  if (type === 'circle')
-    return <CircleIcon styles={styles} size="small" fill={fill} stroke={stroke} />;
-  if (type === 'square')
-    return <SquareIcon styles={styles} size="normal" fill={fill} stroke={stroke} />;
+function icon(step: Flatten<SimpleLegendType['steps']>) {
+  const { stepShape, style, stepIconFill, stepIconStroke, stepName } = step;
+
+  if (stepName === 'Possibly local mappers' || stepName === 'Active mappers') {
```

Comment:
This if should be based not on layer names, but on the fact that the color comes from `style['text-color']` or, better, new type "letter" needs to be added and a PR to dn-be/layers-db to change the legend type for this specific layer